src/sugar/utils.ts

```diff
@@ -1,125 +1,27 @@
-  client: Client,
-  address: string,
-  ledgerVersion?: number,
-): Promise<string> {
-  const request: AccountInfoRequest = {
-    command: 'account_info',
-    account: address,
-    ledger_index: ledgerVersion,
-  }
-  const data = await client.request(request)
-  return dropsToXrp(data.result.account_data.Balance)
-}
-
-// If the marker is omitted from a response, you have reached the end
-async function getRecursiveRecur(
-  getter: Getter,
-  marker: string | undefined,
-  limit: number,
-): Promise<any[]> {
-  const data = await getter(marker, limit)
-  const remaining = limit - data.results.length
-  if (remaining > 0 && data.marker != null) {
-    return getRecursiveRecur(getter, data.marker, remaining).then((results) =>
-      data.results.concat(results),
-    )
-  }
-  return data.results.slice(0, limit)
-}
-
-async function getRecursive(getter: Getter, limit?: number): Promise<any[]> {
-  return getRecursiveRecur(getter, undefined, limit || Infinity)
-}
-
-function renameCounterpartyToIssuer<T>(
-  obj: T & { counterparty?: string; issuer?: string },
-): T & { issuer?: string } {
-  const issuer =
-    obj.counterparty != null
-      ? obj.counterparty
-      : obj.issuer != null
-      ? obj.issuer
-      : undefined
-  const withIssuer = { ...obj, issuer }
-  delete withIssuer.counterparty
-  return withIssuer
-}
-
-export interface RequestBookOffersArgs {
-  taker_gets: Issue
-  taker_pays: Issue
-}
-
-function renameCounterpartyToIssuerInOrder(order: RequestBookOffersArgs) {
-  const taker_gets = renameCounterpartyToIssuer(order.taker_gets)
-  const taker_pays = renameCounterpartyToIssuer(order.taker_pays)
-  const changes = { taker_gets, taker_pays }
-  return { ...order, ..._.omitBy(changes, (value) => value == null) }
-}
-
-async function isPendingLedgerVersion(
-  client: Client,
-  maxLedgerVersion?: number,
-): Promise<boolean> {
-  const response = await client.request({
-    command: 'ledger',
-    ledger_index: 'validated',
-  })
-  const ledgerVersion = response.result.ledger_index
-  return ledgerVersion < (maxLedgerVersion || 0)
-}
-
-async function ensureLedgerVersion(
-  this: Client,
-  options: any,
-): Promise<object> {
-  if (
-    Boolean(options) &&
-    options.ledgerVersion != null &&
-    options.ledgerVersion !== null
-  ) {
-    return Promise.resolve(options)
+import { xAddressToClassicAddress, isValidXAddress } from 'ripple-address-codec'
+
+/**
+ * If an address is an X-Address, converts it to a classic address.
+ *
+ * @param account - A classic address or X-address.
+ * @returns The account's classic address.
+ * @throws Error if the X-Address has an associated tag.
+ */
+export function ensureClassicAddress(account: string): string {
+  if (isValidXAddress(account)) {
+    const { classicAddress, tag } = xAddressToClassicAddress(account)
+
+    // Except for special cases, X-addresses used for requests
+    // must not have an embedded tag. In other words,
+    // `tag` should be `false`.
```

Comment:
Do those special cases use this function, or do they handle it separately?

Reply:
Pretty sure that's talking about transactions, which do handle X-Addresses separately (see `autofill`).

Reply:
Should this function go to `./src/utils` or something. isn't it only uses in `Client.request`?